sklearn/utils/estimator_checks.py

```diff
@@ -1194,6 +1223,7 @@ def check_estimators_fit_returns_self(name, estimator_orig):
     X, y = make_blobs(random_state=0, n_samples=9, n_features=4)
     # some want non-negative input
     X -= X.min()
+    X = gram_matrix_if_pairwise(X, estimator_orig)
```

Comment:
Hmm. Sometimes pairwise is for affinities, sometimes for distances. Estimators requiring distances may not play nicely with affinities and vice-versa. This may not be something we need to deal with now, but @amueller should probably consider an estimator tag which selects between distances and affinities when pairwise. Or we can use the presence of a metric parameter as a heuristic (for now)